src/main/java/com/archisacadeny/course/CourseRepository.java

```diff
@@ -173,4 +175,24 @@ public static int getCourseWithMostStudents() {
         return courseId;
     }
 
+    public static List<Course> getAllCourses() {
+        ArrayList<Course> courses = new ArrayList<>();
+        String query = "SELECT * FROM \"courses\" ";
+
+        try (PreparedStatement statement = DataBaseConnectorConfig.getConnection().prepareStatement(query)) {
+            statement.execute();
+            ResultSet rs = statement.getResultSet();
+
+            while (rs.next()) {
+                courses.add(getCourseById(rs.getInt("id")));
```

Comment:
please don't call sql queries inside a loop like this. Already your fetching a list of courses do manipulation inside the loop do not call different method.